stdlib/public/SDK/Foundation/JSONEncoder.swift

```diff
@@ -2359,11 +2382,29 @@ extension _JSONDecoder {
         }
     }
 
+    fileprivate func unbox<T>(_ value: Any, as type: _JSONStringDictionaryMarker.Type) throws -> T? {
+        var result: Dictionary<String, Any> = [:]
+        let dict = value as! Dictionary<String, NSObject>
```

Comment:
We can't `as!` here as we need to confirm that the value we have does actually match the request. We need to get

```swift
guard let dict = value as? NSDictionary else {
    // throw error; see other type mismatch examples throughout
}
```

or else we'd crash whenever you request a dictionary but something else is present instead.

We can also iterate through the `NSDictionary` directly instead of casting to `[String : NSObject]`; we shouldn't pay the cost of an `O(n)` bridge.

Reply:
Ah, right - I mixed up the value vs. request concepts here.
Also did forgot to consider that the types are already NSDictionaries from the JSONSerialization.

When considering the NSDictionary I also have to ensure that the keys are Strings. Is it ok here to throw a similar type mismatch error on the keys not being Strings?

Reply:
You don't need to worry about this case — `JSONSerialization` can only ever return a dictionary with string keys.